pkg/deploy/generator/resources_rp.go

```diff
@@ -161,9 +166,16 @@ func (g *generator) rpVnet() *arm.Resource {
 				Locations: &[]string{"*"},
 			},
 		}
+	} else {
+		subnet.ServiceEndpoints = &[]mgmtnetwork.ServiceEndpointPropertiesFormat{
+			{
+				Service:   to.StringPtr("Microsoft.Storage"),
```

Comment:
This logic could be cleaner here.  i.e. default to `Microsoft.Storage` service endpoint, and if production then append the other ones.  don't repeat yourself (DRY)

Reply:
Updated -  Changing the order that the service endpoints get added did cause similar changes to the production RP template. This shouldn't matter.

Reply:
@bennerv I'll begin working on a diagram, and then share a link on slack

Reply:
@bennerv Was that diagram I sent over sufficient? Is there anything else you may want or need related to this effort?